modules/ROOT/pages/partner-connector-upgrade.adoc

```diff
@@ -0,0 +1,417 @@
+
+You must also ensure the following MTF dependencies are set in your custom connector’s pom.xml file:
+
+* munit 3.1.0
+* munit-extensions-maven-plugin 1.2.0
+* mtf-tools 1.2.0
+* mule-maven-plugin 4.1.0
+* mule-extensions-maven-plugin 1.6.0-rc1
+
+These MTF dependencies have a minimum Mule version of 4.3.0. To ensure your MTF tests do not validate against Mule runtime versions earlier than 4.3.0, you must add this to the munit-plugin configuration of your custom connector’s pom.xml file: 
+
+[source,xml,linenums]
+----
+<configuration>
+	[...]
+<runtimeConfiguration>
+    <discoverRuntimes>
+        <minMuleVersion>${minVersion}</minMuleVersion>
+        <includeSnapshots>false</includeSnapshots>
+        <product>EE</product>
+    </discoverRuntimes>
+</runtimeConfiguration>
+</configuration>
+----
+
+You can only run MTF tests against Java 17 with Mule runtime 4.6.0 and later. For Mule runtime versions earlier than 4.6.0, you can only run MTF tests against Java 8 and Java 11. 
+
+MUnit 3.1 is only compatible with Mule runtime 4.3.0 and later. If your connector is compatible with Mule runtime 4.2.0 and earlier, you must create a legacy profile that overrides the MUnit version.
+
+=== Read Your Tests 
+
+After you run your MTF tests, your build has either of the following outcomes:
+
+* Test failures
++
+You will likely have to make changes to your custom connector code to ensure Java 17 compatibility.
+
+* All tests pass
++
+Either your custom connector does not require any major changes or your test suite is not comprehensive enough. You might want to review your test suite and double-check that your code coverage is good and that your test scenarios and assertions are not excessively naive.
+
+=== Communicate the Support Level of Your Custom Connector
+
+After you update your code and your tests are green, you are ready to release a new Java 17 compatible version of your custom connector. 
+
+To communicate Java 17 compatibility, you must generate metadata regarding Java compatibility of your custom connector. Add or upgrade the custom connector’s mule-sdk-api dependency to the latest version: 
+
+[source,xml,linenums]
+----
+<dependency>
+   <groupId>org.mule.sdk</groupId>
+   <artifactId>mule-sdk-api</artifactId>
+   <version>0.7.0</version>
+</dependency>
+----
+
+Next, use the new @JavaVersionSupport annotation to specify the added support of Java 17. In Mule 4.5 and later, custom connectors that do not specify the @JavaVersionSupport annotation are assumed to be compatible with Java 8 and Java 11. Add the @JavaVersionSupport annotation in the same class that has the @Extension annotation and include the JAVA_17 value, for example: 
+
+[source,java,linenums]
+----
+@Extension(name = "Database")
+@Operations(...)
+@JavaVersionSupport({JAVA_8, JAVA_11,JAVA_17})
+public class DatabaseConnector {
+..
+}
+----
+
```

Comment:
In Mule 4.5 and later, custom connectors that do not specify the `@JavaVersionSupport` annotation are assumed to be compatible with Java 8 and Java 11.

Reply:
Basically, move this info after the instruction.

Reply:
Mule 4.6 right?